app/components/earthworks/code_snippet_modal_component.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+module Earthworks
+  class CodeSnippetModalComponent < ViewComponent::Base
+    def initialize(document)
+      @document = document
+      @raster_data_type = raster_data?
+      @vector_data_type = vector_data?
+      super
+    end
+
+    def code_block(content)
+      sanitize "<pre><code>#{h content}</code></pre>"
+    end
+
+    def render_python
+      substitute_values(retrieve_text('Python'))
+    end
+
+    def render_r
+      substitute_values(retrieve_text('R'))
+    end
+
+    def render_leaflet
+      substitute_values(retrieve_text('Leaflet'))
+    end
+
+    # Incorporate
+    def substitute_values(content)
+      sub_content = content
+
+      # replace DRUID_ID
+      sub_content = sub_content.gsub('<<WXS_ID>>', @document.wxs_identifier) unless @document.wxs_identifier.empty?
```

Comment:
For all of these substitutions, consider handling the case where the desired field on `@document` is empty.
For example, here if ` @document.wxs_identifier.empty?` is true, the string `<<WXS_ID>>` will remain in the code snippet. This applies to all of the `unless` blocks below within this method as well.

Reply:
I tried to handle all the situations by replacing with a placeholder value where the document value was not present.